### PR TITLE
Babel - Persistent Language Hint for non-primary language

### PR DESCRIPTION
**Pull request by mrschick**
**When merged this pull request will:**
- Add a setting under `ACRE2 UI` that allows for a persistent language hint (it doesn't fade out) whenever a language other than the "main language" (the first assigned) is selected.

### CODE REVIEW

File: addons/sys_list/initSettings.inc.sqf [hunk 1]
```diff
@@ -66,6 +66,15 @@ private _fonts = ["EtelkaMonospacePro", "EtelkaMonospaceProBold", "LCD14", "Puri
     {}
 ] call CBA_fnc_addSetting;
 
+// Babel Language Hint Persistence
+[
+    QGVAR(LanguageHintPersistence),
```
Comment (jonpas) [inline addons/sys_list/initSettings.inc.sqf]:
```suggestion
    QGVAR(LanguageHintPersist),
```
Is a more sensible name.

File: addons/sys_list/stringtable.xml [hunk 2]
```diff
@@ -96,6 +96,16 @@
             <Portuguese>Cor: Alternar Idiomas do Babel</Portuguese>
             <Turkish>Babel Dil Değiştirme Rengi</Turkish>
         </Key>
+        <Key ID="STR_ACRE_sys_list_LanguageHintPersistence_DisplayName">
+            <English>Persistent Babel Language Indicator</English>
+            <German>Babel Sprachanzeige bleibt sichtbar</German>
+            <Italian>Indicatore di lingua persistente</Italian>
+        </Key>
+        <Key ID="STR_ACRE_sys_list_LanguageHintPersistence_Description">
+            <English>When enabled, the Babel Language Indicator in the bottom right will not fade out if a language other than the "main language" (the first assigned spokenLanguage) is being spoken.</English>
```
Comment (jonpas) [inline addons/sys_list/stringtable.xml]:
```suggestion
            <English>Prevent the Babel Language Indicator in the bottom right to fade out if a language other than the "main language" (the first assigned spoken language) is selected.</English>
```
Same for German and Italian.

**Review by jonpas — CHANGES_REQUESTED** [review]

File: addons/sys_list/stringtable.xml (hunk 2 above)
Comment (Mike-MF) [inline addons/sys_list/stringtable.xml]:
```suggestion
            <English>Prevent the Babel Language Indicator in the bottom right from fading out if a language other than the "main language" (the first assigned spoken language) is selected.</English>
```

Reads better in English.

Comment (Drofseh) [reply addons/sys_list/stringtable.xml]: `Prevent the Babel Language Indicator in the bottom right to fade out`
Should be
`Prevent the Babel Language Indicator in the bottom right from fading out`

**Review by Mike-MF — APPROVED** [review]
I tested it, works. Just needs other review addressing.

File: addons/sys_list/stringtable.xml [hunk 3]
```diff
@@ -96,6 +96,16 @@
             <Portuguese>Cor: Alternar Idiomas do Babel</Portuguese>
             <Turkish>Babel Dil Değiştirme Rengi</Turkish>
         </Key>
+        <Key ID="STR_ACRE_sys_list_LanguageHintPersistence_DisplayName">
+            <English>Persistent Babel Language Indicator</English>
```
Comment (jonpas) [inline addons/sys_list/stringtable.xml]:
```suggestion
            <English>Persist Babel Language Indicator</English>
```

File: addons/sys_list/stringtable.xml [hunk 4]
```diff
@@ -96,6 +96,16 @@
             <Portuguese>Cor: Alternar Idiomas do Babel</Portuguese>
             <Turkish>Babel Dil Değiştirme Rengi</Turkish>
         </Key>
+        <Key ID="STR_ACRE_sys_list_LanguageHintPersistence_DisplayName">
```
Comment (jonpas) [inline addons/sys_list/stringtable.xml]:
```suggestion
        <Key ID="STR_ACRE_sys_list_LanguageHintPersist_DisplayName">
```

File: addons/sys_list/stringtable.xml [hunk 5]
```diff
@@ -96,6 +96,16 @@
...
+            <English>Persistent Babel Language Indicator</English>
+            <German>Babel Sprachanzeige bleibt sichtbar</German>
+            <Italian>Indicatore di lingua persistente</Italian>
+        </Key>
+        <Key ID="STR_ACRE_sys_list_LanguageHintPersistence_Description">
```
Comment (jonpas) [inline addons/sys_list/stringtable.xml]:
```suggestion
        <Key ID="STR_ACRE_sys_list_LanguageHintPersist_Description">
```

File: addons/sys_list/initSettings.inc.sqf [hunk 6]
```diff
@@ -66,6 +66,15 @@ private _fonts = ["EtelkaMonospacePro", "EtelkaMonospaceProBold", "LCD14", "Puri
     {}
 ] call CBA_fnc_addSetting;
 
+// Babel Language Hint Persistence
+[
+    QGVAR(LanguageHintPersist),
+    "CHECKBOX",
+    [localize LSTRING(LanguageHintPersistence_DisplayName), localize LSTRING(LanguageHintPersistence_Description)],
```
Comment (jonpas) [inline addons/sys_list/initSettings.inc.sqf]:
```suggestion
    [localize LSTRING(LanguageHintPersist_DisplayName), localize LSTRING(LanguageHintPersist_Description)],
```

File: addons/sys_core/fnc_cycleLanguage.sqf [hunk 7]
```diff
@@ -21,11 +21,18 @@ if (_numSpokenLanguages > 1) then {
     if (_nextId > _numSpokenLanguages - 1) then {
         _nextId = 0;
     };
+
+    // Set hint duration to infinite when you're not speaking your first language (setting dependent)
+    private _hintTime = 1;
+    if (EGVAR(sys_list,LanguageHintPersistence) && {_nextId > 0}) then {
```
Comment (jonpas) [inline addons/sys_core/fnc_cycleLanguage.sqf]:
```suggestion
    if (EGVAR(sys_list,LanguageHintPersist) && {_nextId > 0}) then {
```

**Review by jonpas — CHANGES_REQUESTED** [review]
Setting varname change requires more changes.

**Review by jonpas — APPROVED** [review]

**Pull request merged**